airflow/utils/log/logging_mixin.py

```diff
@@ -18,18 +18,35 @@
 from __future__ import annotations
 
 import abc
+import enum
 import logging
 import re
 import sys
 from io import IOBase
 from logging import Handler, Logger, StreamHandler
-from typing import IO
+from typing import IO, cast
 
 # 7-bit C1 ANSI escape sequences
 ANSI_ESCAPE = re.compile(r"\x1B[@-_][0-?]*[ -/]*[@-~]")
 
-# Private: A sentinel object
-DISABLE_PROPOGATE = object()
+
+# Private: A sentinel objects
+class SetContextPropagate(enum.Enum):
+    """:meta private:"""
+
+    # If a `set_context` function wants to _keep_ propagation set on it's logger it needs to return this
+    # special value.
+    MAINTAIN_PROPAGATE = object()
+    # Don't use this one anymore!
+    DISABLE_PROPAGATE = object()
+
+
+def __getattr__(name):
+    if name in ("DISABLE_PROPOGATE", "DISABLE_PROPAGATE"):
```

Comment:
:D

Reply:
We all knew I couldn't spell right?

(The switch to an Enum was for typing reasons, no easy way of saying "returns none or this _specific_ object type")

Reply:
Looking forward to the depreciation warning hahahaha :D

Reply:
Luckily in this case I don't think we need one: it was only for a single point release, and it didn't really work in practice

😅

Reply:
But I want to see you suffer 😆 You need to learn from those mistakes 😆